common/include/pcl/common/io.h

```diff
@@ -244,13 +244,42 @@ namespace pcl
   PCL_EXPORTS int
   interpolatePointIndex (int p, int length, InterpolationType type);
 
-  /** \brief Concatenate two pcl::PCLPointCloud2.
+  /** \brief Concatenate two pcl::PointCloud<PointT>
+    * \param[in] cloud1 the first input point cloud dataset
+    * \param[in] cloud2 the second input point cloud dataset
+    * \param[out] cloud_out the resultant output point cloud dataset
+    * \return true if successful, false if failed
+    * \ingroup common
+    */
+  template <typename PointT>
+  PCL_EXPORTS bool
+  concatenate (const pcl::PointCloud<PointT> &cloud1,
+               const pcl::PointCloud<PointT> &cloud2,
+               pcl::PointCloud<PointT> &cloud_out)
+  {
+    return pcl::PointCloud<PointT>::concatenate(cloud1, cloud2, cloud_out);
+  }
+
+  /** \brief Concatenate two pcl::PCLPointCloud2
+    * \param[in] cloud1 the first input point cloud dataset
+    * \param[in] cloud2 the second input point cloud dataset
+    * \param[out] cloud_out the resultant output point cloud dataset
+    * \return true if successful, false if failed
+    * \ingroup common
+    */
+  PCL_EXPORTS bool
+  concatenate (const pcl::PCLPointCloud2 &cloud1,
+               const pcl::PCLPointCloud2 &cloud2,
+               pcl::PCLPointCloud2 &cloud_out);
+
+  /** \brief Concatenate two pcl::PCLPointCloud2
     * \param[in] cloud1 the first input point cloud dataset
     * \param[in] cloud2 the second input point cloud dataset
     * \param[out] cloud_out the resultant output point cloud dataset
     * \return true if successful, false if failed (e.g., name/number of fields differs)
     * \ingroup common
     */
+  [[deprecated("use pcl::concatenate() instead")]]
```

Comment:
```suggestion
  [[deprecated("use pcl::concatenate() instead, but beware of subtle difference in behavior (see documentation)")]]
```

This is to prevent users from blindly following the imperative of the deprecation message. And add `\warn` a sentence to the docstring describing the pointing out the difference in behavior with the new function.